students/BrianNisonger/session9/mailroom.py

```diff
@@ -0,0 +1,197 @@
+import statistics
+
+
+class Donor():
+    """ Class used to contain donor name,donation, and methods involved with a single donor """
+
+    def __init__(self, name, donation=""):
+        """ Takes a name, and optionally a donation list"""
+        self.name = name
+        self.donation = list(donation)
+
+    def add_donation(self, *values):
+        """Takes either a single donation, and appends it to the list, or appends a set of donations"""
+        for value in values:
+            self.donation.append(value)
+
+    def make_average(self, value):
+        """Averages the donations"""
+        return statistics.mean(value)
+
+    def make_total(self, value):
+        """Totals the donations"""
+        return sum(value)
+
+    def num_donations(self, value):
```

Comment:
the whole point of a class is that you keep the data (donation history) in the object with the methods that act on it.

so all these should be using self.donations to get the values for this donor.